akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -705,9 +706,9 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
         setHandler(in, new Reading(in, n - pos, getHandler(in))((elem: T) => {
           result(pos) = elem
           pos += 1
-          if (pos == n) andThen(result)
-        }, () => onClose(result.take(pos))))
-      } else andThen(result)
+          if (pos == n) andThen(result.toIndexedSeq)
+        }, () => onClose(result.take(pos).toIndexedSeq)))
+      } else andThen(result.toIndexedSeq)
```

Comment:
This could be a performance regression, creating Vector from the Array each time.

Reply:
Ah yes, should be `a.toSeq`, not `toIndexedSeq` to be identical to the previous implicit conversion, I'll change

Reply:
Or do you think the array wrapper alloc will be troublesome as well? The wrapping should only happen at the end of the n read elements, not for each, but ofc for each call to `readN` it will happen once

Reply:
See my comment to raboof below about this: https://github.com/akka/akka/pull/26673#discussion_r272530721